typescript/packages/plugins/uniswap/src/tools.ts

```diff
@@ -0,0 +1,49 @@
+import type { DeferredTool, EVMWalletClient } from "@goat-sdk/core";
+import type { z } from "zod";
+import { getApprovalTransaction, getQuote, getSwapTransaction, sendTransaction } from "./api";
+import { CheckApprovalBodySchema, GetQuoteBodySchema, GetSwapBodySchema, SendTransactionBodySchema } from "./types";
+
+export type UniswapToolsOptions = {
+    apiKey: string;
+    baseUrl: string;
+};
+
+export function getTools({ apiKey, baseUrl }: UniswapToolsOptions): DeferredTool<EVMWalletClient>[] {
+    return [
+        {
+            name: "check_approval",
+            description:
+                "This {{tool}} checks if the wallet has enough approval for a token and returns the transaction to approve the token. The approval must takes place before the swap transaction.",
+            parameters: CheckApprovalBodySchema,
+            method: async (walletClient: EVMWalletClient, parameters: z.infer<typeof CheckApprovalBodySchema>) => {
+                return getApprovalTransaction(parameters, apiKey, baseUrl);
+            },
+        },
+        {
+            name: "get_quote",
+            description:
+                "This {{tool}} gets the quote for a swap. If permitData is returned, it needs to be signed using the signedTypedData tool.",
+            parameters: GetQuoteBodySchema,
+            method: async (walletClient: EVMWalletClient, parameters: z.infer<typeof GetQuoteBodySchema>) => {
+                return getQuote(parameters, apiKey, baseUrl);
+            },
+        },
+        {
+            name: "get_swap_transaction",
+            description:
+                "This {{tool}} gets the swap transaction for a swap. If permitData was returned from the get_quote tool, it needs to be signed using the signedTypedData tool before calling this function.",
+            parameters: GetSwapBodySchema,
+            method: async (walletClient: EVMWalletClient, parameters: z.infer<typeof GetSwapBodySchema>) => {
+                return getSwapTransaction(parameters, apiKey, baseUrl);
+            },
+        },
+        {
+            name: "send_transaction",
+            description: "This {{tool}} signs and sends a transaction.",
+            parameters: SendTransactionBodySchema,
+            method: async (walletClient: EVMWalletClient, parameters: z.infer<typeof SendTransactionBodySchema>) => {
+                return sendTransaction(parameters, walletClient);
+            },
+        },
+    ];
```

Comment:
had to create this one as I didn't see a wallet plugin that can send a primitive tx

Reply:
A generic sendTx function is quite complex for an LLM to figure out what to do with it.

The goal with these plugins is to have specific functions (e.g swapTokens, createLiquidityPool, collectFees, etc.)

Then these functions can use sendTransaction internally.

I would remove this one. There is another dev adding those specific calls in a different PR so maybe they could contribute to this base with those functions :)

Reply:
That is the case for AI SDK but other AI agent frameworks are not there yet. In any case let's go with this approach! :)

Could you move the sendTransaction method [here](https://github.com/goat-sdk/goat/tree/main/typescript/packages/core/src/plugins) instead? Make it a core plugin instead.